aries_cloudagent/holder/routes.py

```diff
@@ -274,6 +329,114 @@ async def credentials_list(request: web.BaseRequest):
     return web.json_response({"results": credentials})
 
 
+@docs(
+    tags=["credentials"],
+    summary="Fetch DIF credential from wallet by id",
+)
+@match_info_schema(CredIdMatchInfoSchema())
+@response_schema(VCRecordSchema(), 200, description="")
+async def dif_cred_get(request: web.BaseRequest):
+    """
+    Request handler for retrieving DIF credential.
+
+    Args:
+        request: aiohttp request object
+
+    Returns:
+        Verifiable credential record
+
+    """
+    context: AdminRequestContext = request["context"]
+    credential_id = request.match_info["credential_id"]
+
+    session = await context.session()
+    holder = session.inject(VCHolder)
+    try:
+        vc_record = await holder.retrieve_credential_by_given_id(credential_id)
```

Comment:
This fetches the credential by its given id. This is an optional field, and the `id` property in the credential itself. I think `retrieve_credential_by_id` should be used here, and that `given_id` becomes a query parameter for the query credentials.